canonical-web-and-design/maas-ui
# feat(ui): add logs to controllers

## Done

- Display the logs and events for controllers.
- Refactor logs components to work with controllers.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the React logs tab for a controller.
- Check that you can see events and installation output and can download the files.

## Fixes

Fixes: canonical-web-and-design/app-tribe#944.
Fixes: canonical-web-and-design/app-tribe#945.

## Screenshots

<img width="1213" alt="Screen Shot 2022-06-03 at 10 09 24 am" src="https://user-images.githubusercontent.com/361637/171761749-2c850301-d2b0-4d4a-a89b-ad63c3f1799a.png">
